indexing-hadoop/src/main/java/io/druid/indexer/DetermineHashedPartitionsJob.java

```diff
@@ -324,12 +324,9 @@ protected void reduce(
             HyperLogLogCollector.makeCollector(ByteBuffer.wrap(value.getBytes(), 0, value.getLength()))
         );
       }
-      Optional<Interval> intervalOptional = config.getGranularitySpec().bucketInterval(new DateTime(key.get()));
 
-      if (!intervalOptional.isPresent()) {
-        throw new ISE("WTF?! No bucket found for timestamp: %s", key.get());
-      }
-      Interval interval = intervalOptional.get();
+      Interval interval = config.getGranularitySpec().getSegmentGranularity().bucket(new DateTime(key.get()));
```

Comment:
yes. if you look closer in UniformGranularitySpec, you’ll notice that bucket intervals are actually determined by [chopping the input intervals properly according to the granularity spec](https://github.com/druid-io/druid/blob/master/server/src/main/java/io/druid/segment/indexing/granularity/UniformGranularitySpec.java#L62)
so whatever `bucket()` computes should align with the interval from `bucketInterval()` as well when it returns one b/c the goal of both is to return an interval that follows the rules of the granularity.